diff -ur rts5229/Makefile rts5229_new/Makefile
--- rts5229/Makefile	2012-03-21 02:40:18.000000000 +0100
+++ rts5229_new/Makefile	2016-01-21 21:15:17.173225848 +0100
@@ -22,7 +22,7 @@
 # Makefile for the PCI-Express Card Reader drivers.
 #
 
-TARGET_MODULE := rts5229
+TARGET_MODULE := rts5227
 
 EXTRA_CFLAGS := -Idrivers/scsi 
 
diff -ur rts5229/rtsx.c rts5229_new/rtsx.c
--- rts5229/rtsx.c	2012-03-21 02:40:18.000000000 +0100
+++ rts5229_new/rtsx.c	2016-01-21 21:25:56.906856635 +0100
@@ -143,9 +143,9 @@
 
 	
 	SPRINTF("       Vendor: Realtek Corp.\n");
-	SPRINTF("      Product: RTS5229\n");
+	SPRINTF("      Product: RTS5227\n");
 	SPRINTF("      Version: %s\n", DRIVER_VERSION);
-	SPRINTF("        Build: %s, %s\n", __DATE__, __TIME__);
+	//SPRINTF("        Build: %s, %s\n", __DATE__, __TIME__);
 
 	/*
 	 * Calculate start of next buffer, and return value.
@@ -263,7 +263,7 @@
 	
 	.name =				CR_DRIVER_NAME,
 	.proc_name =			CR_DRIVER_NAME,
-	.proc_info =			proc_info,
+	//.proc_info =			proc_info,
 	.info =				host_info,
 
 	
@@ -518,14 +518,14 @@
 		 * the maximum known LUN
 		 */
 		else if (chip->srb->device->id) {
-			printk(KERN_ERR "Bad target number (%d:%d)\n",
-				  chip->srb->device->id, chip->srb->device->lun);
+			//printk(KERN_ERR "Bad target number (%d:%d)\n",
+			//	  chip->srb->device->id, chip->srb->device->lun);
 			chip->srb->result = DID_BAD_TARGET << 16;
 		}
 
 		else if (chip->srb->device->lun > chip->max_lun) {
-			printk(KERN_ERR "Bad LUN (%d:%d)\n",
-				  chip->srb->device->id, chip->srb->device->lun);
+			//printk(KERN_ERR "Bad LUN (%d:%d)\n",
+			//	  chip->srb->device->id, chip->srb->device->lun);
 			chip->srb->result = DID_BAD_TARGET << 16;
 		}
 
@@ -911,14 +911,14 @@
 	chip->support_mmc = 1;
 }
 
-static int __devinit rtsx_probe(struct pci_dev *pci, const struct pci_device_id *pci_id)
+static int  rtsx_probe(struct pci_dev *pci, const struct pci_device_id *pci_id)
 {
 	struct Scsi_Host *host;
 	struct rtsx_dev *dev;
 	int err = 0;
 	struct task_struct *th;
 
-	printk(KERN_INFO "--- %s, %s ---\n", __DATE__, __TIME__);
+	//printk(KERN_INFO "--- %s, %s ---\n", __DATE__, __TIME__);
 
 	err = pci_enable_device(pci);
 	if (err < 0) {
@@ -1066,7 +1066,7 @@
 }
 
 
-static void __devexit rtsx_remove(struct pci_dev *pci)
+static void  rtsx_remove(struct pci_dev *pci)
 {
 	struct rtsx_dev *dev = (struct rtsx_dev *)pci_get_drvdata(pci);
 
@@ -1092,7 +1092,7 @@
 	.name = CR_DRIVER_NAME,
 	.id_table = rts5229_ids,
 	.probe = rtsx_probe,
-	.remove = __devexit_p(rtsx_remove),
+	.remove = rtsx_remove,
 #ifdef CONFIG_PM
 	.suspend = rtsx_suspend,
 	.resume = rtsx_resume,
diff -ur rts5229/rtsx.h rts5229_new/rtsx.h
--- rts5229/rtsx.h	2012-03-21 02:40:18.000000000 +0100
+++ rts5229_new/rtsx.h	2016-01-21 21:16:02.249053489 +0100
@@ -40,6 +40,7 @@
 #include <linux/workqueue.h>
 #include <linux/timer.h>
 #include <linux/time.h>
+#include <linux/vmalloc.h>
 
 #include <scsi/scsi.h>
 #include <scsi/scsi_cmnd.h>
@@ -52,7 +53,7 @@
 #include "trace.h"
 #include "general.h"
 
-#define CR_DRIVER_NAME		"rts5229"
+#define CR_DRIVER_NAME		"rts5227"
 
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 14)
